exercises/ciara_nightingale/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,118 @@
+#include <algorithm>
+#include <ctype.h>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+#include <sstream>
+
+// replaces the hypens with whitespace - used in clean_word
+bool replace_hyphens(std::string& word) {
+	bool test = false; //so the whole word is checked rather than stopping at the first instance of a hyphen
+	for (std::string::size_type i = 0; i != word.size(); ++i) {
+		//check through each character in the word
+		if (word[i] == '-') {
+				word[i] = ' ';
+				test = true;
+		}
+  }
+	if (test == true) {
+			return true;
+	}
+  return false;
+}
+
+// determine if a character is punctuation - used in clean_word
+bool is_punc(char const c) {
+  static const std::string punc = ".,?\'\"!();:";
+  for (std::string::size_type i = 0; i != punc.size(); ++i) {
+    if (c == punc[i]) {
+      return true;
+      break;
+    }
+  }
+  return false;
+}
+
+// deal with punctuation and case sensitivity - used in process_word
+void clean_word(std::string& word) {
+  // remove characters if they are puctuation
+  word.erase(remove_if(word.begin(), word.end(), is_punc), word.end());
+  // convert characters to lowercase if they are uppercase
+  transform(word.begin(), word.end(), word.begin(), tolower);
+}
+
+//clean the words, count them and put them into the map
+void processWord(std::string& word, std::map<std::string, int>& counter, std::string::size_type& maxlen, std::string::size_type const min_word_length) {
+		clean_word(word);
+		maxlen = std::max(maxlen, word.size()); //find the longest word.
+		if (word.size() > min_word_length) {
+				++counter[word];
+				// add the word to the map and the no. of appearances in the text.
+		}
+}
+
+// convert a map to a vector of pairs
+std::vector<std::pair<std::string, int>> const
+mapToVector(const std::map<std::string, int> &map) {
+  return std::vector<std::pair<std::string, int>>(map.begin(), map.end());
+}
+
+// determine which pair is greater
+bool compare(std::pair<std::string, int> const pair_1,
```

Comment:
This function name could be more descriptive, `compare` could be doing any number of things. Something more specific would improve readability when the function is called somewhere you can't see its docstring. 

Naming can be hard, and this is a tricky example, but maybe something along the lines of `isGreater` or `pairValueIsGreater`